fix(*): "We"/"our"/etc. to "you"/"your"/etc. (#661)
discordjs/guide: guide/popular-topics/audit-logs.md

2) There is no event which triggers when an audit log is created.
:::

Let's start by glancing at the `fetchAuditLogs` method and how to work with it. Like many discord.js methods, it returns a Promise containing the GuildAuditLogs object. In most cases, only the `entries` property will be of interest, as it holds a collection of GuildAuditLogsEntry objects, and consequently, the information you usually want. You can always take a look at the options <docs-link path="class/Guild?scrollTo=fetchAuditLogs">in the discord.js docs</docs-link>.

The following examples will explore a straightforward case for some auditLog types. Some basic error handling is performed, but these code segments are by no means foolproof and are meant to teach you how fetching audit logs work. You will most likely need to expand on the examples based on your own goals for a rigorous system.

## Who deleted a message?

> One of the most common usages for audit logs would be understanding who deleted any given message in a Discord server.

Reviewer: ```suggestion
One of the most common use cases for audit logs would be understanding who deleted any given message in a Discord server.
```

::: warning
At the time of writing, Discord does not emit an audit log if the person who deleted the message is a bot deleting a single message or is the author of the message itself.
:::

For now, we'll focus on the `messageDelete` event.

```js
client.on('messageDelete', message => {
	console.log(`A message by ${message.author.tag} was deleted, but we don't know by who yet.`);
});
```

So far, nothing should seem new or complicated. You get the message deleted event and log that a message was removed from a channel. More information from the message object can be extracted, but that is left as an exercise for the reader.

For simplicity, set a fetch limit of 1 and accept only the `MESSAGE_DELETE` type.

> Placing this into the previous code, you get the following. Note that this also makes the function async to make use of `await`. In addition, be make sure to ignore DMs.

Reviewer 2: ```suggestion
Placing this into the previous code, you get the following. Note that this also makes the function async to make use of `await`. In addition, make sure to ignore DMs.
```

		limit: 1,
		type: 'MESSAGE_DELETE',
	});
	// Since there's only 1 audit log entry in this collection, grab the first one
	const deletionLog = fetchedLogs.entries.first();

	// Perform a coherence check to make sure that there's *something*
	if (!deletionLog) return console.log(`A message by ${message.author.tag} was deleted, but no relevant audit logs were found.`);

	// Now grab the user object of the person who deleted the message
	// Also grab the target of this action to double-check things
	const { executor, target } = deletionLog;

	// Update the output with a bit more information

> 	// Also run a check to make sure the log returned was for the same author's message

Reviewer 2: IMO this sounds better with `returned`
```suggestion
	// Also run a check to make sure that the log returned was for the same author's message
```

});
```

The same as before: set the fetch limit to 1 and accept only the `MEMBER_KICK` type.

```js {2-7,9-10,12-14,16-22}
client.on('guildMemberRemove', async member => {
	const fetchedLogs = await member.guild.fetchAuditLogs({
		limit: 1,
		type: 'MEMBER_KICK',
	});
	// Since there's only 1 audit log entry in this collection, grab the first one
	const kickLog = fetchedLogs.entries.first();

	// Perform a coherence check to make sure that there's *something*
	if (!kickLog) return console.log(`${member.user.tag} left the guild, most likely of their own will.`);

	// Now grab the user object of the person who kicked the member
	// Also grab the target of this action to double-check things
	const { executor, target } = kickLog;

	// Update the output with a bit more information

> 	// Also run a check to make sure the log returned was for the same kicked member

Reviewer 2: Same as above
```suggestion
	// Also run a check to make sure that the log returned was for the same kicked member
```

```js
client.on('guildBanAdd', async (guild, user) => {
	console.log(`${user.tag} got hit with the swift hammer of justice in the guild ${guild.name}.`);
});
```

As was the case in the previous examples, you can see what happened, to whom it happened, but not who executed the action. Enter once again audit logs fetching limited to 1 entry and only the `MEMBER_BAN_ADD` type. The `guildBanAdd` listener then becomes:

```js {2-7,9-10,12-14,16-22}
client.on('guildBanAdd', async (guild, user) => {
	const fetchedLogs = await guild.fetchAuditLogs({
		limit: 1,
		type: 'MEMBER_BAN_ADD',
	});
	// Since there's only 1 audit log entry in this collection, grab the first one
	const banLog = fetchedLogs.entries.first();

	// Perform a coherence check to make sure that there's *something*
	if (!banLog) return console.log(`${user.tag} was banned from ${guild.name} but no audit log could be found.`);

	// Now grab the user object of the person who banned the member
	// Also grab the target of this action to double-check things
	const { executor, target } = banLog;

	// Update the output with a bit more information

> 	// Also run a check to make sure the log returned was for the same banned member

Reviewer 2: Same as above
```suggestion
	// Also run a check to make sure that the log returned was for the same banned member
```
